### DISCUSSION TITLE
Connect to Cisco Devices

### DISCUSSION THREAD

**Discussion by Admirall777** [Eugeny/tabby#9822] [General]
Tabby Version:1.0.209
Cisco 3750 version 12.2
When use "diffie-hellman-group1-sha1" - connection close with "Unknown DH group".
When use other algorithm - connection close with "Handshake failed: no matching key exchange algorithm".
Whot can i do for open SSH session to the device?
Ciphers aes256-cbc
HMAC hmac-sha1
Host key ssh-rsa

**Comment by Eugeny** [comment]
Check out https://github.com/Eugeny/tabby/issues/9810

**Comment by Admirall777** [comment]
Thank you wery match. Nightly version is really work with Cisco. But they still don't work with ViPNet normally (https://github.com/Eugeny/tabby/discussions/9785). May be you know how i can fix it?
And one more: Huawei commutators need "diffie-hellman-group-exchange-sha1" and in nightly don't connect :(
